### PR TITLE
memhp_threads: Add check for prealloc-threads default

### PR DESCRIPTION
**Pull request by zhenyzha**
Add check for prealloc-threads default

ID: 2140499
Signed-off-by: zhenyzha <zhenyzha@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
@YongxueHong @mcasquer  Could you pls help review when you are free. thanks.

git am 3537.patch
Applying: memhp_threads: Add check for prealloc-threads default

```
on aarch64 qemu7.1:
(1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.memhp_threads.arm64-pci: PASS (43.62 s)
on aarch64 qemu7.0:
 (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.memhp_threads.arm64-pci: PASS (54.32 s)

on x86 qemu7.1:
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.memhp_threads.q35: PASS (113.52 s)

on x86 qemu7.0:
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.memhp_threads.q35: PASS (126.74 s)
```

**Review by mcasquer — APPROVED** [review]
Tested on x86_64:
```bash
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.memhp_threads.q35: STARTED
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.memhp_threads.q35: PASS (73.60 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

LGTM mcasquer@redhat.com

File: qemu/tests/memhp_threads.py [hunk 1]
```diff
@@ -41,34 +44,45 @@ def get_qemu_threads(cmd, timeout=60):
...
-    dimm = vm.devices.dimm_device_define_by_params(params.object_params(mem),
-                                                   mem)
-    dimm.set_param("memdev", dev["id"])
-    test.log.info("Hotplug pc-dimm '%s' to guest", dimm["id"])
-    vm.devices.simple_hotplug(dimm, vm.monitor)
-    memhp_test.update_vm_after_hotplug(vm, dimm)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    qemu_version = utils_qemu.get_qemu_version(qemu_binary)[0]
+    if qemu_version in VersionInterval('[7.1.0,)'):
+        threads_default = int(params["smp"])
```
Comment (YongxueHong) [inline qemu/tests/memhp_threads.py:51]:
```suggestion
        threads_default = params.get_numeric("smp")
```

File: qemu/tests/memhp_threads.py [hunk 2]
```diff
@@ -41,34 +44,45 @@ def get_qemu_threads(cmd, timeout=60):
...
+        dev = Memory(new_params["backend"], new_params)
+        dev.set_param("id", "%s-%s" % ("mem", target_mem))
+        args = [vm.monitor, vm.devices.qemu_version]
+        bg = utils_test.BackgroundTest(dev.hotplug, args)
+        test.log.info("Hotplug memory backend '%s' to guest", dev["id"])
+        bg.start()
+        if target_mem == "plug1":
+            threads_num = threads_default
+        else:
+            threads_num = int(params["prealloc-threads_mem_plug2"])
```
Comment (YongxueHong) [inline qemu/tests/memhp_threads.py:71]:
```suggestion
            threads_num = params.get_numeric("prealloc-threads_mem_plug2")
```

File: qemu/tests/memhp_threads.py (hunk 1 above)
Comment (zhenyzha) [reply qemu/tests/memhp_threads.py]: thanks a lot, has been corrected.

File: qemu/tests/memhp_threads.py (hunk 2 above)
Comment (zhenyzha) [reply qemu/tests/memhp_threads.py]: thanks a lot, has been corrected.

**Comment by zhenyzha** [comment]
@YongxueHong Could you pls help review again when you are free? thanks.

**Comment by zhenyzha** [comment]
The update has been done  @YongxueHong Could you pls help review it again when you are free? thanks.

git am 3537.patch
Applying: memhp_threads: Add check for prealloc-threads default

` (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.memhp_threads.arm64-pci: PASS (118.20 s)`

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**